lib/galaxy/datatypes/qiime2.py

```diff
@@ -76,6 +92,46 @@ def sniff(self, filename):
         return metadata and metadata['semantic_type'] == 'Visualization'
 
 
+class PredicateRemover(ast.NodeTransformer):
+    def visit_BinOp(self, node):
+        if isinstance(node.op, ast.Mod):
+            return PredicateRemover().visit(node.left)
+        return node
+
+
+class ReconstructExpression(ast.NodeVisitor):
+    expression = '%s'
+    tuple_count = 0
+    in_index = False
+
+    def visit_Name(self, node):
+        if self.tuple_count == 0:
+            self.expression = self.expression % node.id
+        else:
+            self.expression = self.expression % node.id + ', %s'
+            self.tuple_count -= 1
+
+        self.generic_visit(node)
+
+    def visit_Index(self, node):
+        pre_strip = len(self.expression)
+        self.expression = self.expression.rstrip(', %s')
+        post_strip = len(self.expression)
+
+        self.expression += '[%s'
```

Comment:
Something to consider, you could probably just skip the string interpolation since it appears to always happen on the right, even in this case. So then yourdefault expression could be `''` and you would just add to it.

Reply:
Yeah I think we are at a point where the interpolation is no longer needed.